homeassistant/components/risco/__init__.py

```diff
@@ -90,6 +90,9 @@ async def _async_setup_local_entry(hass: HomeAssistant, entry: ConfigEntry) -> b
 
     async def _error(error: Exception) -> None:
         _LOGGER.error("Error in Risco library", exc_info=error)
+        if isinstance(error, ConnectionResetError) and not hass.is_stopping:
+            _LOGGER.debug("Disconnected from panel. Reloading integration")
+            hass.async_create_task(hass.config_entries.async_reload(entry.entry_id))
```

Comment:
Why would we completely reload the integration instead of just reconnecting?

Reply:
We don't know if reconnect will be successful. Reloading the integration leverages the retry mechanism, as well as accurately reflects the status.
I modeled this after the Matter and ZHA integrations, after a conversation on Discord.

Reply:
Thanks for clarification 👍